### ISSUE TITLE
Windows Support?

### ISSUE THREAD

**Issue by DJuego** [oatpp/oatpp#2]
Hi! It seems a interesting resource. Thank you for your efforts, @lganzzzo !

However i am interested in a cross platform solution. So  are Windows  (and Mac Os) supported by oat++?

Thanks again.

DJuego

**Comment by lganzzzo** [comment]
Hey, thank you for your interest in the project! @DJuego 

- **Mac OS** is supported out of the box.
- **Windows** is not currently supported out of the box. However, since oatpp is _connection agnostic_ it is quite _easy to add windows support_.
All you need to do is define your own **ConnectionProvider** utilizing Windows sockets.
You may take a look at https://github.com/oatpp/oatpp/blob/master/network/server/SimpleTCPConnectionProvider.hpp
for example.

Also contributions to oatpp for windows support are very much appreciated!

Best Regards,
lganzzzo

**Comment by lganzzzo** [comment]
Hey All,

All the latest news on this issue are here - https://github.com/oatpp/oatpp/issues/93

**Comment by codderembedded** [comment]
Hi, 
i saw it can be installed on windows. Is this true? I tryed but it wasn't. Error LNK1181 cannot open input file 'Release\oatpp.lib'
Error LNK1181 cannot open input file '..\src\Release\oatpp-test.lib'

**Comment by bhorn** [comment]
Hi, this looks more like a misconfigured library path or build system rather than an oatpp problem.
Can you elaborate more on what you have tried or have done so we can might come up with what went wrong? A mere linker error does not say very much.

**Comment by codderembedded** [comment]
I download oatpp and I watched this stages on windows. I run 'developer command prompt for VS 2019 as administrator.
$ git clone https://github.com/oatpp/oatpp.git
$ cd oatpp\
$ MD build
$ cd build\

and I download CMake. I chose the file path where oatpp is located and I I created the build file in this file. And configure and generate. Then oatpp -> bulid  I opened oatpp.sln with visual studio. and all build. then i encountered the this error
